ddtrace/tracer/option_test.go

```diff
@@ -466,29 +463,6 @@ func TestTracerOptionsDefaults(t *testing.T) {
 	})
 }
 
-func TestDefaultHTTPClient(t *testing.T) {
-	t.Run("no-socket", func(t *testing.T) {
-		// We care that whether clients are different, but doing a deep
-		// comparison is overkill and can trigger the race detector, so
-		// just compare the pointers.
-		assert.Same(t, defaultHTTPClient(), defaultClient)
-	})
-
-	t.Run("socket", func(t *testing.T) {
-		f, err := ioutil.TempFile("", "apm.socket")
-		if err != nil {
-			t.Fatal(err)
-		}
-		if err := f.Close(); err != nil {
-			t.Fatal(err)
-		}
-		defer os.RemoveAll(f.Name())
-		defer func(old string) { defaultSocketAPM = old }(defaultSocketAPM)
-		defaultSocketAPM = f.Name()
-		assert.NotSame(t, defaultHTTPClient(), defaultClient)
```

Comment:
⚠️ Why was this test removed? The new `TestHTTPClient/UDS` test that replaces it is nice (it actually tests socket communication), but it only verifies that this works for `DD_TRACE_AGENT_URL`. The default socket discovery is no longer tested.